superset/views/core.py

```diff
@@ -650,6 +650,10 @@ def explore(  # pylint: disable=too-many-locals,too-many-return-statements
                 f"datasource_id={datasource_id}&"
             )
 
+        # if feature enabled, run some health check rules for sqla datasource
+        if hasattr(datasource, "health_check") and conf["DATASET_HEALTH_CHECK"]:
```

Comment:
I think you can skip the conf check here. It will be immediately checked called by `datasource.health_check()` anyway.